spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/examples/SparkLogModuleOptionsMetadata.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.spark.streaming.examples;
+
+import org.springframework.xd.module.options.spi.ModuleOption;
+import org.springframework.xd.spark.streaming.DefaultSparkStreamingModuleOptionsMetadata;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+public class SparkLogModuleOptionsMetadata extends DefaultSparkStreamingModuleOptionsMetadata {
```

Comment:
Sorry by "hidden from the user", I actually meant "not a concern for the developer". In other words, if a developer is providing a Processor impl (thus providing a custom XD module that uses Spark Streaming), we want them to have the simplest possible experience. In the case where you only need the default options, you would simply add DefaultSparkStreamingModuleOptionsMetadata to the properties file. If you need other options in addition to those, you would provide SomeModuleOptionsMetadata impl of your own. It seems that using "extends" there is pretty simple, but if you think there's a simpler option *from that module developer's perspective*, we should consider it.